include/roboteam_ai/stp/invariants/NoGoalVisionFromBallInvariant.h

```diff
@@ -0,0 +1,38 @@
+//
+// Created by timovdk on 5/27/20.
+//
+
+#ifndef RTT_NOGOALVISIONFROMBALLINVARIANT_H
+#define RTT_NOGOALVISIONFROMBALLINVARIANT_H
+
+#include <NFParam/Param.h>
+
+#include "BaseInvariant.h"
+
+namespace rtt::ai::stp::invariant {
+/**
+    Calculates goal vision percentage based on the position of the ball
+**/
+class NoGoalVisionFromBallInvariant : public BaseInvariant {
+   public:
+    NoGoalVisionFromBallInvariant() noexcept;
+
+    [[nodiscard]] uint8_t metricCheck(world_new::view::WorldDataView world, const world::Field* field) const noexcept override;
+
+   private:
+    /**
+     * Calculates the actual metric value using the piecewise linear function member
+     * @param x the x of the function
+     * @return metric value between 0-255
```

Comment:
What does this metric value represent? That is currently unclear from your description.

Reply:
If you look at the documentation in the base class, it might become clearer